esphome/components/pmsx003/pmsx003.cpp

```diff
@@ -173,163 +175,143 @@ optional<bool> PMSX003Component::check_byte_() {
-
-      uint16_t pm_1_0_concentration = this->get_16_bit_uint_(10);
-      uint16_t pm_2_5_concentration = this->get_16_bit_uint_(12);
-      uint16_t pm_10_0_concentration = this->get_16_bit_uint_(14);
-
-      uint16_t pm_particles_03um = this->get_16_bit_uint_(16);
-      uint16_t pm_particles_05um = this->get_16_bit_uint_(18);
-      uint16_t pm_particles_10um = this->get_16_bit_uint_(20);
-      uint16_t pm_particles_25um = this->get_16_bit_uint_(22);
-      // Note the pm particles 50um & 100um are not returned,
-      // as PMS5003T uses those data values for temperature and humidity.
-
-      float temperature = (int16_t) this->get_16_bit_uint_(24) / 10.0f;
-      float humidity = this->get_16_bit_uint_(26) / 10.0f;
-
-      ESP_LOGD(TAG,
-               "Got PM1.0 Concentration: %u µg/m^3, PM2.5 Concentration %u µg/m^3, PM10.0 Concentration: %u µg/m^3, "
-               "Temperature: %.1f°C, Humidity: %.1f%%",
-               pm_1_0_concentration, pm_2_5_concentration, pm_10_0_concentration, temperature, humidity);
-
-      if (this->pm_1_0_std_sensor_ != nullptr)
-        this->pm_1_0_std_sensor_->publish_state(pm_1_0_std_concentration);
-      if (this->pm_2_5_std_sensor_ != nullptr)
-        this->pm_2_5_std_sensor_->publish_state(pm_2_5_std_concentration);
-      if (this->pm_10_0_std_sensor_ != nullptr)
-        this->pm_10_0_std_sensor_->publish_state(pm_10_0_std_concentration);
-
-      if (this->pm_1_0_sensor_ != nullptr)
-        this->pm_1_0_sensor_->publish_state(pm_1_0_concentration);
-      if (this->pm_2_5_sensor_ != nullptr)
-        this->pm_2_5_sensor_->publish_state(pm_2_5_concentration);
-      if (this->pm_10_0_sensor_ != nullptr)
-        this->pm_10_0_sensor_->publish_state(pm_10_0_concentration);
-
-      if (this->pm_particles_03um_sensor_ != nullptr)
-        this->pm_particles_03um_sensor_->publish_state(pm_particles_03um);
-      if (this->pm_particles_05um_sensor_ != nullptr)
-        this->pm_particles_05um_sensor_->publish_state(pm_particles_05um);
-      if (this->pm_particles_10um_sensor_ != nullptr)
-        this->pm_particles_10um_sensor_->publish_state(pm_particles_10um);
-      if (this->pm_particles_25um_sensor_ != nullptr)
-        this->pm_particles_25um_sensor_->publish_state(pm_particles_25um);
-
-      if (this->temperature_sensor_ != nullptr)
-        this->temperature_sensor_->publish_state(temperature);
-      if (this->humidity_sensor_ != nullptr)
-        this->humidity_sensor_->publish_state(humidity);
-      break;
-    }
+    if (this->formaldehyde_sensor_ != nullptr)
+      this->formaldehyde_sensor_->publish_state(formaldehyde);
+  }
+
+  // Temperature and Humidity
+  if (this->type_ == PMSX003_TYPE_5003ST || this->type_ == PMSX003_TYPE_5003T) {
+    const uint8_t temperature_offset = (this->type_ == PMSX003_TYPE_5003T) ? 24 : 30;
+
+    const float temperature = (int16_t) this->get_16_bit_uint_(temperature_offset) / 10.0f;
```

Comment:
```suggestion
    const float temperature = static_cast<int16_t>(this->get_16_bit_uint_(temperature_offset)) / 10.0f;
```
Preferred over C-style cast because it is more explicit and safer, also easier to find with Ctrl+F